src/basic/log.c

```diff
@@ -1357,6 +1358,9 @@ int log_syntax_internal(
                 const char *func,
                 const char *format, ...) {
 
+        if (level == 3 || level == 4)
```

Comment:
What is "3", what is "4" — use appropriate symbolical constants.

Why is `error_check` set here? This operation does not match the intent of the variable described in the comment above.